core/standards.md

```diff
@@ -0,0 +1,80 @@
+# Core Smart Contracts
+
+## Definition of Core
+
+A core contract is one that:
+
+1. Any contract that is part of the [dss repo](https://github.com/makerdao/dss)
+2. Will have ward status on or other privileged access to any other core contract (i.e. `Vat.rely(<new contract address>)`)
+3. Will rely on the Pause Proxy as a permissioned actor. If the pause proxy has to directly interact with this contract then there is additional security and process work that needs to be done.
+4. Will be included in the chainlog as a canonical “Maker Contract”. Even if the contract does not have elevated permissions in the system, a failure of a contract in the chainlog could result in cost (bug bounties) or reputation damage for the protocol.
+5. If a contract is put forward to a business partner of the DAO, by DAO members, close consideration should be given to whether it would be perceived as core Maker brand by those partners.
+
+## Protocol Engineering Review Process for Core Contracts
+
+All Core contracts need to go through a risk adjusted review process.  Risk should be evaluated at the beginning of the project and continue to be re-evaluated through the engineering process.
+
+Our review process at a minimum requires two team member reviews and approvals.  However for higher risk contracts may also involve Red Team reviews and audits by external auditors.
+
+Core contracts should follow established coding patterns including repo and contract structures, variable naming patterns (yes Daiwanese is a standard for our team) and auth patterns.
+
+Whenever possible, utilize an existing solution instead of building a new contract. MakerDAO is a very complex project and each new contract or derivation increases the cognitive and maintenance overhead of the entire protocol. By limiting the number of moving parts we reduce the risk of unexpected behavior.
+
+This repo attempts to document our standards, other sources include (non-exhaustive list):
+
+- [MIPS best practices](https://github.com/makerdao/mips-best-practices)
+
+## Coding standards we expect to see in core contracts
+
+### Permissions
+
+Permissioning in the contract should be handled by `wards` and the `wards` mapping should occupy storage slot `0x0`
+
+### Contract ordering
+
+Following and expanding on [Solidity Order of layout recommendations](https://docs.soliditylang.org/en/stable/style-guide.html#order-of-layout) our contracts are generally laid out in this way:
+
+1. State variables (including wards and math)
```

Comment:
On this topic the solidity version also goes up the top so if we are being particular with that we should mention it here?

Reply:
I feel this mostly describes the actual contract, not the file.  If we expand to the contract file, then probably need to include:

- line break at end
- pragma version
- license
- one contract per file
etc

Reply:
I think it would be good to have something on this.

If we take a contract where all the above conditions were satisfied (a line at the end, the pragma version was incorrect and it was multiple contracts per file / multi file contract) we would probably not be too happy about it! 

I agree that this probably wasn't the best section of the document to raise it, but if it's something we would push back on if it didn't match our standards, IMO we should make our standards known and accessible.